tangelo/toolboxes/molecular_computation/molecule.py

```diff
@@ -313,9 +313,18 @@ def mo_coeff(self):
     @mo_coeff.setter
     def mo_coeff(self, new_mo_coeff):
         # Asserting the new molecular coefficient matrix have the same dimensions.
-        assert self.solver.mo_coeff.shape == (new_mo_coeff := np.array(new_mo_coeff)).shape, \
-            f"The new molecular coefficients matrix has a {new_mo_coeff.shape}"\
-            f" shape: expected shape is {self.solver.mo_coeff.shape}."
+        if self.uhf:
+            for j in range(2):
+                new_mo_coeff = list(new_mo_coeff)
```

Comment:
This does not depend on `j` and is not modified in the loop: are you sure this instruction should be inside the loop or is needed at all ?

There's no docstring with the type of `new_mo_coeff` here so I'm not sure what it is and why it needs to be cast as a list. I think we should have some sort of doc.